modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSClient.java

```diff
@@ -223,7 +223,9 @@ private WFSStrategy determineCorrectStrategy() {
             } else if (uri.toLowerCase().contains("/arcgis/services/")
                     && Versions.v2_0_0.equals(capsVersion)) {
                 strategy = new ArcGisStrategy_2_0();
-            } else if (uri.contains("mapserver") || uri.contains("map=")) {
+            } else if ((uri.contains("mapserver") || uri.contains("map="))
```

Comment:
As indicated in the comment, the change has been added because the MapServerWFSStrategy is a WFS 1.0 and 1.1 strategy that won't work with WFS 2.0 (and besides, it's accounting for old MapServer bugs). 
Hence the extra check. For WFS 2.0 we end up using the general WFS 2.0 strategy instead.

Reply:
Ok, yeah, I looked up the Jira issue Gabriel was quoted from and that adds a little more context.  Sounds like this is an old (and entrenched) problem.